code/datums/components/toggle_coat.dm

```diff
@@ -0,0 +1,47 @@
+/datum/component/toggle_coat
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	var/buttoned = null
+	var/coat_style = null
+	var/obj/item/clothing/suit/suit = null
+	var/obj/ability_button/coat_toggle/toggle = new
+
+/datum/component/toggle_coat/Initialize(buttoned = null, coat_style = null)
+	if(!istype(parent, /obj/item/clothing/suit))
+		return COMPONENT_INCOMPATIBLE
+	src.buttoned = buttoned
+	src.coat_style = coat_style
+	src.suit = parent
+	if (!islist(suit.ability_buttons))
+		suit.ability_buttons = list()
+	suit.ability_buttons += toggle
```

Comment:
`LAZYLISTADD` can replace all this. Doesn't seem to be well documented but it's a pretty comprehensible macro.